libc-top-half/musl/include/dlfcn.h

```diff
@@ -7,24 +7,40 @@ extern "C" {
 
 #include <features.h>
 
+#ifdef __wasilibc_unmodified_upstream
 #define RTLD_LAZY   1
 #define RTLD_NOW    2
 #define RTLD_NOLOAD 4
 #define RTLD_NODELETE 4096
 #define RTLD_GLOBAL 256
 #define RTLD_LOCAL  0
+#else
+/* For WASI, we use flag values which match MacOS rather than musl.  This gives
+ * `RTLD_LOCAL` a non-zero value, avoiding ambiguity and allowing us to defer
+ * the decision of whether `RTLD_LOCAL` or `RTLD_GLOBAL` should be the default
+ * when neither is specified.
+ */
+#define RTLD_LAZY     0x1
+#define RTLD_NOW      0x2
+#define RTLD_LOCAL    0x4
+#define RTLD_GLOBAL   0x8
+#define RTLD_NOLOAD   0x10
+#define RTLD_NODELETE 0x80
```

Comment:
Why re-define all of these just because you want to make RTLD_LOCAL non-zero?

Reply:
We don't have to.  I just figured we should try to consistently match _some_ existing system that is already widely used, and MacOS was the first one I found that had a non-zero RTLD_LOCAL.  I don't have any strong feelings about it, though.  Would you prefer I change only RTLD_LOCAL?

Reply:
I think we should always prefer the minimal set of changes from musl... unless there is a good reason not to.

Reply:
I just pushed another update that uses the musl values except for RTLD_LOCAL.